staging/src/k8s.io/apimachinery/pkg/util/manifest/manifest.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"bytes"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// NewUnstructuredList decodes YAML or JSON data into a flattened list of Kubernetes resources,
+// it will attempt parsing all manner of manifests, such as YAML multi docs and any list types
```

Comment:
I'm a little skeptical that many people will want to flatten multidocs with mixed lists and loose objects. I recommend making this two steps, e.g. read the multidoc into a list, then optionally call a "flatten" function to expand embedded lists. But I'd like to see a use case for that latter step before putting it in a general library, since it's easy to add stuff here and impossible to remove it :)

Reply:
Here's a very simple example:
```
---
apiVersion: v1
kind: List
items:
- apiVersion: v1
   kind: Namespace
   metadata:
       name: my-namespace
```
Having parsed this to `unstructured.UnstructuredList` you have no way of reasoning about the depth and it's too easy to forget that it can be deeply nested, or even no too deeply as per above example.

I agree that it doesn't have a separate explicit step or a a boolean function argument would be a more sensible way to implement this.

The main challenge is really - how does a user read all objects from a manifest and iterate over them to pass them to a client, and they very little about internal representation, but they are familiar with `k8s.io/api`, and they might presume that there's some type that holds abstract objects (like `unstructured.Unstructured`).

Reply:
@lavalamp I've changed it to not call `AppendFlattened`

Reply:
Comment needs an update now :)